src/lib.rs

```diff
@@ -1497,3 +1627,18 @@ impl SwapChain {
         }
     }
 }
+
+struct OwnedLabel(Option<CString>);
```

Comment:
Note that I haven't actually tested this yet but I think the pointer lifetime is ok. Also interested in ways to improve this

Reply:
I have a fully tested inlinable label mechanism here that we didn't end up using in wgpu-core.

https://github.com/aloucks/wgpu/commit/ca02eba2a7d9106b3e0ccaadb467072f32e025a2

Reply:
That's a useful mechanism indeed. I think it's fine to own `OwnedLabel` here for it being simple enough, and then follow-up.